components/brave_today/rust/lib.rs

```diff
@@ -0,0 +1,120 @@
+extern crate feed_rs;
+use lazy_static::lazy_static;
+use feed_rs::parser;
+use regex::Regex;
+
+#[cxx::bridge(namespace = brave_news)] mod ffi {
+  pub struct FeedItem {
+    id: String,
+    title: String,
+    description: String,
+    image_url: String,
+    destionation_url: String,
+    published_timestamp: i64,
+  }
+
+  pub struct FeedData {
+    id: String,
+    title: String,
+    items: Vec<FeedItem>
+  }
+
+  extern "Rust" {
+    fn parse_feed_string(source: String, output: &mut FeedData) -> bool;
+  }
```

Comment:
How common are non-utf-8 rss documents? `feed-rs` explicitly takes a type implementing `Read` so it can do its own charset detection/decoding, while `String` guarantees valid utf-8 encoding. `::rust::String()` says it throws if that validation fails, so probably we'd see some combination of silent failure and silent text corruption in those cases.

Probably you want `source: &[u8]` here. You can pass `::rust::Slice<const uint8_t>(body_content.data(), body_content.size())` on the C++ size (modulo signedness casting)

Reply:
I'll try this as a follow-up, thanks. Logged here: https://github.com/brave/brave-browser/issues/20592